### DISCUSSION TITLE
Why doesn't the array sort?

### DISCUSSION THREAD

**Discussion by jereklas** [mobxjs/mobx#3788] [General] [answered]
I'm wondering if someone could explain how the assertion in the array test case is valid?

https://github.com/mobxjs/mobx/blob/main/packages/mobx/__tests__/v5/base/array.js#L129

Grabbing the value of the computed in the autorun should have caused the array to become sorted again and the value should be [1,2,3]?

**Comment by mweststrate** [comment, marked as answer]
The test is that the computed throws when trying to do so :). `ar.sort()`
is by design forbidden on an observable array.

**Reply by jereklas** [reply]
Ah thanks! I completely misread the test and thought it was one where it was checking for a console log but still allowing the operation to occur.
